src/Compilers/VisualBasic/Portable/Binding/Binder_Statements.vb

```diff
@@ -3118,9 +3120,15 @@ Namespace Microsoft.CodeAnalysis.VisualBasic
             If targetTypeIsValid Then
                 initialValue = ApplyImplicitConversion(initialValue.Syntax, targetType, initialValue, diagnostics)
                 limit = ApplyImplicitConversion(limit.Syntax, targetType, limit, diagnostics)
+                Dim stepValueBeforeConversion = stepValue
                 stepValue = ApplyConversion(stepValue.Syntax, targetType, stepValue,
                                             isExplicit:=forStatement.StepClause Is Nothing,
                                             diagnostics:=diagnostics)
+
+                If stepValue IsNot stepValueBeforeConversion AndAlso stepValue.Kind = BoundKind.Conversion AndAlso
+                   forStatement.StepClause Is Nothing Then
```

Comment:
Is it sufficient to check `forStatement.StepClause Is Nothing`, or perhaps `stepValueBeforeConversion.WasCompilerGenerated()`?

Reply:
I hesitate doing this on arbitrary nodes because it is not allowed to mark nodes as compiler generated after this state was checked by someone.